src/libraries/System.Private.CoreLib/src/System/SpanHelpers.ByteMemOps.cs

```diff
@@ -0,0 +1,538 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+#if TARGET_AMD64 || TARGET_ARM64 || (TARGET_32BIT && !TARGET_ARM) || TARGET_LOONGARCH64
+// JIT is guaranteed to unroll blocks up to 64 bytes in size
+#define HAS_CUSTOM_BLOCKS
+#endif
+
+using System.Diagnostics;
+using System.Numerics;
+using System.Runtime.CompilerServices;
+using System.Runtime.InteropServices;
+
+namespace System
+{
+    internal static partial class SpanHelpers // .ByteMemOps
+    {
+#if TARGET_ARM64 || TARGET_LOONGARCH64
+        private const ulong MemmoveNativeThreshold = ulong.MaxValue;
+#elif TARGET_ARM
+        private const nuint MemmoveNativeThreshold = 512;
+#else
+        private const nuint MemmoveNativeThreshold = 2048;
+#endif
+        // TODO: Determine optimal value
+        private const nuint ZeroMemoryNativeThreshold = 1024;
+
+
+#if HAS_CUSTOM_BLOCKS
+        [StructLayout(LayoutKind.Sequential, Size = 16)]
+        private struct Block16 {}
+
+        [StructLayout(LayoutKind.Sequential, Size = 64)]
+        private struct Block64 {}
+#endif // HAS_CUSTOM_BLOCKS
+
+#if NATIVEAOT
+        [System.Runtime.RuntimeExport("RhSpanHelpers_MemCopy")]
+#endif
+        [Intrinsic] // Unrolled for small constant lengths
+        internal static unsafe void Memmove(ref byte dest, ref byte src, nuint len)
+        {
+            // P/Invoke into the native version when the buffers are overlapping.
+            if (((nuint)(nint)Unsafe.ByteOffset(ref src, ref dest) < len) || ((nuint)(nint)Unsafe.ByteOffset(ref dest, ref src) < len))
```

Comment:
```suggestion
            if (((nuint)Unsafe.ByteOffset(ref src, ref dest) < len) || ((nuint)Unsafe.ByteOffset(ref dest, ref src) < len))
```
ByteOffset returns IntPtr that is an alias for nint these days. The `nint` casts should not be needed, I think.